### PR TITLE
[ci, batch2] fix batch2 secrets part 1

### PR DESCRIPTION
**Pull request by cseed**
First of 3 changes, I think:
 - batch2 secret field now requires namespace and it is used for lookup (it was required before in validation, but ignored)
 - CI now sets namespace consistently.  Since batch1 pods can only mount secrets from batch-pods, it only sets this to batch-pods.  The one exception is when the secretes come from a runImage step.  Then the namespace is taken from the runImage step if it is present, and otherwise defaults to batch-pods.

Two more PRs:
1. Set the namespace in all runImage step secrets
2. Make the namespace required in runImage steps

Then, when we switch CI from batch1 => batch2, we can create runImage steps that use secrets from namespaces other than batch-pods.

### CODE REVIEW

**Comment by cseed** [comment]
Note, in a few instances, this changed batch2 to pull secrets from batch-pods rather than default to match the behavior of batch1.  I think this might break some stuff.  I will debug as needed.

File: batch2/deployment.yaml [hunk 1]
```diff
@@ -39,8 +39,10 @@ spec:
            value: /deploy-config/deploy-config.json
          - name: BATCH_WORKER_IMAGE
            value: "{{ batch2_worker_image.image }}"
-         - name: BATCH_NAMESPACE
+         - name: HAIL_DEFAULT_NAMESPACE
            value: "{{ default_ns.name }}"
+         - name: HAIL_BATCH_PODS_NAMESPACE
```
Comment (jigold) [inline batch2/deployment.yaml]: These also have to be added below. There's two deployment configs here and I think you only got one of them.

Comment (cseed) [reply batch2/deployment.yaml]: Fixed, thanks!  Might be more issues...

File: batch2/service-account-batch-pods.yaml [hunk 2]
```diff
@@ -0,0 +1,21 @@
+kind: Role
```
Comment (jigold) [inline batch2/service-account-batch-pods.yaml]: How is this file different than `batch2/service-account.yaml`?

Comment (jigold) [reply batch2/service-account-batch-pods.yaml]: Is this enabling batch2 to also get things from batch-pods?

**Comment by jigold** [comment]
It's getting a forbidden error when trying to download the secret. I think I know why. The service account being used is the `batch2` service account from the default namespace. I think we need to do what I did with CI where the service account lives in the default namespace, but it can read secrets, service accounts in the batch pods namespace.

File: batch2/service-account-batch-pods.yaml (hunk 2 above)
Comment (jigold) [reply batch2/service-account-batch-pods.yaml]: I understand now. Disregard please.

**Comment by jigold** [comment]
Hmmm.... You already do this. I'm going to try to be more explicit to make sure that's not the problem.

**Comment by cseed** [comment]
OK, for the record, I changed ci-agent to:
 - only exist in batch-pods,
 - have deploy permissions in both default and batch-pods.

In the future, I think we should:
 - have a service account just for creating and destroying namespace (creator-and-destroyer-of-namespaces) used by create namespace step,
 - the deploy step should use the admin account for the target namespace,
 - and similarly for create database.

This means we'll eventually need to include the namespace with the service account in batch.

**Pull request merged**